server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTargetService.java

```diff
@@ -84,6 +86,37 @@ public void beforeIndexShardClosed(ShardId shardId, @Nullable IndexShard indexSh
         }
     }
 
+    /**
+     * Invoked when a new checkpoint is received from a primary shard.
+     * It checks if a new checkpoint should be processed or not and starts replication if needed.
+     * @param requestCheckpoint       received checkpoint that is checked for processing
+     * @param indexShard      replica shard on which checkpoint is received
+     */
+    public synchronized void onNewCheckpoint(final ReplicationCheckpoint requestCheckpoint, final IndexShard indexShard) {
```

Comment:
Nitpick - use parameter names `receivedCheckpoint` and `replicaShard` for clarity

Reply:
Sure will do this in next commit